ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -553,12 +566,40 @@ public void applyIndexedAttestations(final List<ValidateableAttestation> attesta
               attestations.stream()
                   .map(this::getIndexedAttestation)
                   .forEach(
-                      attestation -> forkChoiceStrategy.onAttestation(transaction, attestation));
+                      attestation ->
+                          forkChoiceStrategy.onAttestation(
+                              transaction,
+                              attestation,
+                              recentChainData.getStore().getEquivocatingIndices()));
               transaction.commit();
             })
         .reportExceptions();
   }
 
+  public void onAttesterSlashing(
+      final AttesterSlashing slashing,
+      InternalValidationResult validationStatus,
+      boolean fromNetwork) {
+    if (!equivocatingIndicesEnabled || !validationStatus.isAccept()) {
+      return;
+    }
+    final Checkpoint retrievedJustifiedCheckpoint =
+        recentChainData.getStore().getJustifiedCheckpoint();
+    final StoreTransaction transaction = recentChainData.startStoreTransaction();
+    recentChainData
+        .retrieveCheckpointState(retrievedJustifiedCheckpoint)
+        .thenPeek(
+            justifiedCheckpointState ->
+                onForkChoiceThread(
+                        () ->
+                            slashing
+                                .getIntersectingValidatorIndices()
+                                .forEach(transaction::addEquivocatingIndex))
+                    .reportExceptions())
+        .thenPeek(__ -> transaction.commit().join())
+        .reportExceptions();
```

Comment:
We should be starting the transaction only after we've retrieved the checkpoint state and ensure we're on the fork choice thread for the whole lifetime of the transaction.

That said, I don't see why we actually need the `justifiedCheckpointState` here. Nothing is actually using it.